src/Datagrid/PagerInterface.php

```diff
@@ -99,4 +99,6 @@ public function setMaxPageLinks(int $maxPageLinks): void;
      * Returns the maximum number of page numbers.
      */
     public function getMaxPageLinks(): int;
+
+    public function isDeterministic(): bool;
```

Comment:
This is a good idea but it's a BC break to add a method in an interface.

You have to do like this
https://github.com/sonata-project/SonataAdminBundle/blob/672a614bd96fac8b633a07571eff4ee2fd3db934/src/Filter/FilterInterface.php#L22-L25

By just adding a `@method` and it will be mandatory in next major.